javascript/index.js

```diff
@@ -160,58 +159,64 @@ function decode(paymentReq){
     const tags = [];
     const final={};
     const fin_content={};
+    const unknowns={};
     final['string']=paymentRequest;
     final['type']=type;
     final['valid']='True'//Need to verify offer_ID with signature.
-    const tlv=[];
-    // console.log(TAGPARSER)
-    // console.log(words_8bit)
-    while(words_8bit.length){
-        let tlvs=''
-	// FIXME: This is a BigSize not an u8.
-        const tagCode = words_8bit[0]
-        tlvs += Buffer.from(words_8bit.slice(0,1)).toString('hex')
-	// FIXME: this is perfectly legal.
+    buffer= (Buffer.from(words_8bit))
+    
+    while(buffer.length){
+        let tlvs=[];
+
+        let res=fromwire_bigsize(buffer);
+        
+        const tagCode=res[0];
+        
+        tlvs.push(Number(''+tagCode));
```

Comment:
This coersion is now unnecessary I think: fromwire_bigsize() returns an integer.